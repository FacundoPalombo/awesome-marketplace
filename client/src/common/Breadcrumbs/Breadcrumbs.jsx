import PropTypes from "prop-types";
import React from "react";
import styles from "./Breadcrumbs.module";

export function Breadcrumbs({ categories }) {
  return (
    <nav className={styles.breadcrumbs}>
      {categories?.length &&
        categories.map((category, index) => (
          <span key={category.id} className={styles.container}>
            <a className={styles.breadcrumb}>{category.name}</a>
            {categories.length - 1 !== index && " > "}
          </span>
        ))}
    </nav>
  );
}

Breadcrumbs.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({ name: PropTypes.string, id: PropTypes.string })
  ),
};
