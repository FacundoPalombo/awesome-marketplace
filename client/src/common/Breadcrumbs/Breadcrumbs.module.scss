@import "utils/variables.scss";

.breadcrumbs {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  text-overflow: ellipsis;
}

.container {
  display: flex;
  align-items: center;
  margin-right: 0.5em;
  justify-content: space-around;
  text-overflow: ellipsis;
}

.breadcrumb {
  display: flex;
  margin-right: $unit;
  width: 100%;
  max-height: 34px;
  white-space: nowrap;
  font-size: 14px;
  text-overflow: ellipsis;
  @include md {
    font-size: 12px;
  }
}
