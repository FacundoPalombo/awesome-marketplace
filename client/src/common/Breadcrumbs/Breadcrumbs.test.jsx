import React from "react";
import { Breadcrumbs } from "./Breadcrumbs";
import { render } from "@testing-library/react";

describe("GIVEN Breadcrumb component", () => {
  describe("WHEN rendered", () => {
    it("THEN should match snapshot", () => {
      const categoriesFixture = [
        {
          id: "MLA1182",
          name: "Instrumentos Musicales",
          results: 21327,
        },
        {
          id: "MLA1747",
          name: "Repuestos Autos y Camionetas",
          results: 44080,
        },
      ];
      const { asFragment } = render(
        <Breadcrumbs categories={categoriesFixture} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
