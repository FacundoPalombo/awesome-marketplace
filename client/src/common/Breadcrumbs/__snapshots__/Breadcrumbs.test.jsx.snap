// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Breadcrumb component WHEN rendered THEN should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumbs"
  >
    <span
      class="container"
    >
      <a
        class="breadcrumb"
      >
        Instrumentos Musicales
      </a>
       &gt; 
    </span>
    <span
      class="container"
    >
      <a
        class="breadcrumb"
      >
        Repuestos Autos y Camionetas
      </a>
    </span>
  </nav>
</DocumentFragment>
`;
